api/models/Members.js

```diff
@@ -36,6 +36,25 @@ var hash_password = key => (member, cb) => {
   })
 }
 
+var update_membership = (member, cb) => {
+  Members
+    .findOne(member.id)
+    .exec(function (error, item) {
+      if (error) {
+        return sails.log.error(error)
+      } else if (member.membership_type && (member.membership_type !== item.membership_type)) {
```

Comment:
We don't need the `else if` here because of the `return` above.

I'd prefer not to have one big function that does everything. Rather, I think I'd do something like

```js
var handle_membership_change = (member, cb) =>
  Members
    .findOne(member.id)
    .exec(function (error, item) {
      if (error) return sails.log.error(error)
      // you could also do the opposite check and only need one `cb(member)` call after the `if`
      if (!member.membership_type || member.membership_type === item.membership_type) return cb(member)
      // if you're going to mutate
      member.date_membership_type_changed = new Date()
      member.life_payment_date = member.membership_type.match('life') ? new Date() : null
      cb(member)
    })
```

and `beforeUpdate: (member, cb) => update_membership(member, member => hash_password(member, cb))`

With a bit of tweaking you could get `beforeUpdate` neater / pointfree

:)